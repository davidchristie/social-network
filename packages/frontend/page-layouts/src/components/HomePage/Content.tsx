import { Container, Section } from "design-system";
import React from "react";

const Content: React.StatelessComponent = () => {
  return (
    <Container className="HomePage">
      <Section>
        <h1>Home</h1>
      </Section>
    </Container>
  );
};

export default Content;
