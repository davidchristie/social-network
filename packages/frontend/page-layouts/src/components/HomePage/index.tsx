import React from "react";
import Page from "../Page";

const HomePage = () => {
  return <Page load={() => import("./Content")} />;
};

export default HomePage;
