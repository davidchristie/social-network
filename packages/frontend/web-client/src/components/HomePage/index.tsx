import { Container, Section } from "design-system";
import React from "react";

export default class HomePage extends React.Component {
  public render () {
    return (
      <Container className="HomePage">
        <Section>
          <h1>Home</h1>
        </Section>
      </Container>
    );
  }
}
