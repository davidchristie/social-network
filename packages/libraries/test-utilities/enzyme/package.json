{
  "main": "../dist/enzyme/index.js",
  "name": "test-utilities/enzyme",
  "private": true,
  "version": "1.0.0"
}
