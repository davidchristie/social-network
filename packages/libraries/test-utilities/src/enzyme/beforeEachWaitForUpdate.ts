import { Wrapper } from "./types";

export default function beforeEachWaitForUpdate (getWrapper: () => Wrapper) {
  beforeEach(async () => {
    await new Promise(resolve => window.setTimeout(resolve, 0));
    getWrapper().update();
  });
}
