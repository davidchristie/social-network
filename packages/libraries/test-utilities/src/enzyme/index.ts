export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
