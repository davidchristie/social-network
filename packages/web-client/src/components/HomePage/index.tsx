import React from "react";

import Section from "../Section";

export default class HomePage extends React.Component {
  public render () {
    return (
      <div>
        <Section>
          <h1>Home</h1>
        </Section>
      </div>
    );
  }
}
