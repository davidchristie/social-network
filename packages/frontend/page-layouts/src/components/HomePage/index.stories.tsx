import React from "react";
import { createStories } from "storybook-utilities";
import HomePage from ".";

createStories("Page Layouts").add(
  "HomePage",
  () => <HomePage />
);
