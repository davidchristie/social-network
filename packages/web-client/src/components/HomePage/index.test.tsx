import { shallow } from "enzyme";
import React from "react";

import HomePage from "./index";

describe("HomePage component", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper).toMatchSnapshot();
  });
});
