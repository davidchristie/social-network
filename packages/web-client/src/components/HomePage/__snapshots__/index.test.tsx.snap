// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component matches snapshot 1`] = `
<div>
  <Section>
    <h1>
      Home
    </h1>
  </Section>
</div>
`;
